task/sql-tasks.js

```diff
@@ -280,7 +473,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT
+        Customers.CompanyName as CompanyName,
+        Products.ProductName as ProductName,
+        OrderDetails.UnitPrice as PricePerItem
+    FROM Customers
+    INNER JOIN Orders ON Orders.CustomerID = Customers.CustomerID
+    INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+    WHERE OrderDetails.UnitPrice = (
+        SELECT 
+            MAX(OD.UnitPrice)
+        FROM Customers as C
+        INNER JOIN Orders AS O ON C.CustomerID = O.CustomerID 
+        INNER JOIN OrderDetails AS OD ON O.OrderID = OD.OrderID
+        WHERE C.CompanyName = Customers.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.